### PR TITLE
[Enhancement][API] Copy workflow with scheduler info

### LINKED ISSUE #10671: [Enhancement][API] Suggest support batch copy workflow with scheduler
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Should the batch copy workflow support copy scheduler of workflow

### Use case

The workflow and the scheduler of the workflow should be copied together so that the copied workflow does not have to reconfigure the scheduler

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuxt2015**
## Purpose of the pull request
fix #10671

## Brief change log
 batch copy workflow with timing configuration

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Review by SbloodyS — DISMISSED** [review]
LGTM

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 1]
```diff
@@ -1800,6 +1802,17 @@ protected void doBatchOperateProcessDefinition(User loginUser,
                     }
                     processDefinition.setLocations(JSONUtils.toJsonString(jsonNodes));
                 }
+                //copy timing configuration
+                Schedule scheduleObj = scheduleMapper.queryByProcessDefinitionCode(oldProcessDefinitionCode);
+                if (scheduleObj != null) {
+                    scheduleObj.setProcessDefinitionCode(processDefinition.getCode());
+                    scheduleObj.setReleaseState(ReleaseState.OFFLINE);
+                    int insertResult = scheduleMapper.insert(scheduleObj);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:1868]: I think you should set id to 0 before insert

Comment (zhuxt2015) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:1868]: Not necessarily, id will auto increment

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 2]
```diff
@@ -1788,6 +1788,7 @@ private Map<String, Object> checkParams(User loginUser,
         return result;
     }
 
+    @Transactional(rollbackFor = RuntimeException.class)
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java]: `this.doBatchOperateProcessDefinition`  will make the `@Transactional` invalidated, you can hold a reference of ProcessDefinitionServiceImpl and use the reference to call doBatchOperateProcessDefinition.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java [hunk 3]
```diff
@@ -1856,6 +1858,19 @@ protected void doBatchOperateProcessDefinition(User loginUser,
                     }
                     processDefinition.setLocations(JSONUtils.toJsonString(jsonNodes));
                 }
+                //copy timing configuration
+                Schedule scheduleObj = scheduleMapper.queryByProcessDefinitionCode(oldProcessDefinitionCode);
+                if (scheduleObj != null) {
+                    scheduleObj.setProcessDefinitionCode(processDefinition.getCode());
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:1864]: Please set the processDefinitionName to the new Name.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java (hunk 3 above)
Comment (ruanwenjun) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java:1864]: Ignore this, I find processDefinitionName is not a column in `t_ds_schedules`.

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Review by ruanwenjun — APPROVED** [review]

**Pull request merged**